packages/databyss-ui/editor/slate/page/ContentEditable.js

```diff
@@ -272,8 +293,15 @@ const SlateContentEditable = forwardRef(
       if (hotKeys.isEsc(event)) {
         onNewBlockMenu(false, editor)
       }
+
+      if (isInlineSourceSelected(editor) && event.key === 'Enter') {
```

Comment:
Ok, I see why we need to pass `onEditSource` into `ContentEditable`. 
This event handler is higher level than the one inside `EditorInline`, so let's pass it into `renderInline` so all source edit actions are passing through the same higher-level function.
When we add topic editing, we'll want to abstract this handler (and related code) into something like `onEditEntity` so we're not repeating ourselves.

Reply:
I attempted this and got suck on passing extra properties to `renderInline`. Lets discuss this next time we talk.